react/wbooks/src/app/components/BookDetail/index.js

```diff
@@ -47,6 +34,9 @@ class BookDetailContainer extends Component {
 }
 
 BookDetailContainer.propTypes = {
+  book: shape({ publicationYear: string, editorial: string, genre: string, imageUrl: string }).isRequired,
```

Comment:
define this in the general `propTypes` folder as `bookPropTypes.js` and export this and a array propType for this...
also all internal shape properties are required, but the shape don't.
```
export const bookPropTypes = shape({ publicationYear: string.isRequired, editorial: string.isRequired, genre: string.isRequired, imageUrl: string.isRequired });

export const booksPropTypes = arrayOf(bookPropTypes);
```

Reply:
I dont know what prop type means with required (and the doc is not clear)
I put that the shape is required because the initial state of book is { }. I thought that not required means null or undefined, not { }

Reply:
And I put the props not required because when I have the inital book { }, there is no property. So if I put them required I think it will give me an error

Reply:
I didnt dig enough in the investigation since using typescript there is no need for this not too happy library.
And my plan is to use typescript! :D